compiler/ilgen/MethodBuilder.cpp

```diff
@@ -80,16 +86,22 @@ namespace OMR
 
 MethodBuilder::MethodBuilder(TR::TypeDictionary *types, OMR::VirtualMachineState *vmState)
    : TR::IlBuilder(asMethodBuilder(), types),
+   // Note: Any allocations in the following memory region that will get deallocated by C++ destruction semantics along with this MethodBuilder class, should instead
+   // be explicitly cleared / deallocated in the MethodBuilder destructor *before* the region itself is destroyed in order to avoid undefined behaviour from invalid
+   // memory references.
```

Comment:
Sorry, I don't understand the comment. It says "Any allocations in the following memory region" which sounds a lot like the std::map's that are allocated as part of this MethodBuilder object. But you must not mean those allocations because the destructor below doesn't explicitly clear / deallocate those std::maps.

After the conversation earlier today, I was expecting a description along the lines of: because the memory region is stored here as a pointer, we can't rely on the usual C++ destruction semantic to destruct and deallocate the memory region  only after the objects allocated from it have been destructed. Also, because the impact is really in the destructor, I expected the comment to be in the destructor rather than in the constructor.

Special points if you explain that the memory region is stored as a pointer to avoid increasing the number of header files needed to compile against the JitBuilder library.